WIP: make a start on some db docs (#192)
DiamondLightSource/SynchWeb: api/docs/db/DataCollections.md

wavelength | float | Wavelength of beamline in angstrom
detectordistance | float | Detector distance in mm
resolution | float | Resolution at edge of detector in angstrom
transmission | float | Transmission in %, i.e. 10
xtalSnapshotFullPath1 | path | Path to a snapshot of the crystal at 0
xtalSnapshotFullPath2 | path | Path to a snapshot of the crystal at 90 (for gridscan with grid and results annotated)
xtalSnapshotFullPath3 | path | Path to a snapshot of the crystal at 180 (for gridscans with grid annotated)
phistart | float | For multi-axis goniomnetry, phi angle at start
kappastart | float | For multi-axis goniometry, kappa angle at start
omegastart | float | For multi-axis goniometry, omega angle at start
chistart | float | For multi-axis goniometry, chi angle at start
beamsizeatsamplex | float | Beamsize at the sample in microns in horizontal
beamsizeatsampley | float | Beamsize at the sample in microns in vertical
xbeam | float | Position of beam in horizontal
ybeam | float | Position of beam in vertical
detectorid | int | A detectorid from `Detector`

Multiple data collections can be stored in the same group, generally this applies to things like wedged SAD collections.

> 

Reviewer: GitHub appears not to like two blank lines before a header:
```suggestion
```

## Types

Data collections are classified using some fairly cryptic column combinations. These should be consolidated and simplified in the long term.

### Screening

A screening data collection is inferred from `DataCollection.overlap` > 0

### Grid Scans

`DataCollectionGroup.experimentType` should be populated as `Mesh`, although grid is actually inferred from:
```
DataCollection.numberOfImages > 1 && DataCollection.axisrange = 0
```
In the long term this should be consolidated to use `DataCollectionGroup.experimentType`

> For gridscans the third crystal snapshot is used `DataCollection.xtalsnapshotfullpath3`. Interally this is because `xtalsnapshotfullpath1` is of the crystal, `xtalsnapshotfullpath2` has the grid with results annotated, and `xtalsnapshotfullpath3` has the grid alone. 

Reviewer: ```suggestion
For gridscans the third crystal snapshot is used `DataCollection.xtalsnapshotfullpath3`. Internally this is because `xtalsnapshotfullpath1` is of the crystal, `xtalsnapshotfullpath2` has the grid with results annotated, and `xtalsnapshotfullpath3` has the grid alone. 
```

datacollectionid | int | A  `DataCollection` id
dx_mm | int | Step size in x in mm for each step in the grid. Not the total size of the grid.
dy_mm | int | Step size in y in mm for each step in the grid.
steps_x | int | Number of steps in x in the grid scan
steps_y | int | Number of steps in y in the grid scan
snapshot_offsetxpixel | int | Number of pixels from top left of the snapshot image in x to where the top left of the grid starts
snapshot_offsetypixel | int | Number of pixels from top left of the snapshot image in y to where the top left of the grid starts
pixelspermicronx | float | Confusingly, microns per pixel. Physical size of the pixel in x in microns
pixelspermicrony | float | Confusingly, microns per pixel. Physical size of the pixel in y in microns

This allows the snapshot to be cropped to the correct location and the grid to be navigatable by click to show an resulting diffraction image. The results from Per Image Analysis are then merged on top as a heatmap.

### Oscillation Data Collection

Data collection is assumed when `DataCollection.axisrange` > 0 and `DataCollection.overlap` = 0

> 

Reviewer: GitHub appears not to like two blank lines before a header:
```suggestion
```

# Detector

`Detector` can be populated, this will allow the diffraction image viewer to correctly calculate resolution limits, etc.

Column | Type | Description
--- | --- | ---
detectortype | str | Type of detector, integrating, photon counter, ...
detectormanufacturer | str | Detector manufacturer
detectormodel | str | Detector model
detectorpixelsizehorizontal | float | Size of pixels in microns in horizontal
detectorpixelsizevertical | float | Size of pixels in microns in vertical
detectorserialnumber | str | Serial number of detector
numberofpixelsx | int | Number of pixels in x
numberofpixelsy | int | Number of pixels in y

# Per Image Analysis

> Per image analysis is populated into `ImageQualityIndicators`. The is calculated for a selection of 250 images during normal data collection and for every image for GridScans

Reviewer: ```suggestion
Per image analysis is populated into `ImageQualityIndicators`. These are calculated for a selection of up to 250 images during normal oscillation data collections and for every image for grid scans.
```
